var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -114,13 +111,27 @@ def install(self, spec, prefix, install_src=True):
         if install_src:
             self._install_src(prefix)
 
-    @staticmethod
-    def _patch_special(prefix, libname='libnrnmech.so'):
-        """Patch bash-based special to point to nrnmech lib inside lib/
-        """
-        which('sed')('-i',
-                     's#-dll .*#-dll %s "$@"#' % prefix.lib.join(libname),
+    def _install_binaries(self, lib_suffix=''):
+        # Install special
+        arch = self.spec.architecture.target
+        prefix = self.prefix
+        shutil.copy(join_path(arch, 'special'), prefix.bin)
+
+        # Install libnrnmech - might have several links.
+        for f in find(arch + "/.libs", 'libnrnmech*.so*', recursive=False):
+            if not os.path.islink(f):
+                bname = os.path.basename(f)
+                lib_dst = prefix.lib.join(bname[:bname.find(".")] + lib_suffix + ".so")
+                shutil.move(f, lib_dst)
+                break
```

Comment:
Does this work on OSX where there is no `.so` files by `.dylib`?

Reply:
Also in osx these files are .so, because they're not 100% shared libs, but libtool "modules".